Corrected error relating already existing Job, Sensor state and Run Id for sesnor

abdulbasitds opened the pull request:
Make sure you have checked _all_ steps below.

### JIRA
- [ ] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-XXX
    - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a JIRA issue.


### Description
- [ ] Here are some details about my PR, including screenshots of any UI changes:


### Tests
- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:


### Commits
- [ ] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"


### Documentation
- [ ] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.


### Code Quality
- [ ] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

OElesin committed:
Added AWS Glue Job Compatibility to Airflow

OElesin committed:
[AIRFLOW-2310] : Added AWS Glue Job Compatibility to Airflow

OElesin committed:
[AIRFLOW-2310] : Added AWS Glue Job Compatibility to Airflow

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow. Fixes Travis build Error

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow. Fixes Travis build Error

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow. Fixed flake8 errors

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests; fixed bug

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests; Also added Glue Job Operator and Glue Job Sensor

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests; Also added Glue Job Operator and Glue Job Sensor

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests; Updated ASF headers

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests; clean ups

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests; clean ups

OElesin committed:
[AIRFLOW-2310] : Add AWS Glue Job Compatibility to Airflow with tests

OElesin committed:
[AIRFLOW-2310]: AWS Glue Job Integration with tests

OElesin committed:
[AIRFLOW-2310]: Added Hook For AWS Glue

OElesin committed:
[AIRFLOW-2310]: Add AWS Glue Integration to Apache Airflow

abdulbasitds committed:
Adding JobRunId to task_instance, so it can be provided to sesnor

In sesnor, you can retrieve as 

AwsGlueJobSensor(task_id='test_glue_job_sensor',
                              job_name='testjob',
                              run_id="{{task_instance.xcom_pull(task_ids='glue_job11', key='run_id')}}",
                              poke_interval=1,
                              timeout=5,
                              aws_conn_id='aws_default',
                                  dag=dag)

abdulbasitds committed:
Retrieving task_instance from context

abdulbasitds committed:
Added/Corrected aready existing job function, Corrected job_completion state fetch error

Added/Corrected aready existing job function,It was not properly checking that if the job already exist or not. To launch already existing job it was trying to create new one and was giving error

Corrected job_completion state fetch was giving error as it was only fetching state once before the while loop and not after once it enter while loop, so was remaining in running state forever

abdulbasitds committed:
Job_state is a Dictionary not state itself, corrected error of calling upper() on it

job_completion was returning and storing dictionary to job_state so need ot access write key job_state['JobRunState'] and then apply .upper() instead of job_state.upper() which was giving error

mik-laj commented on airflow/contrib/hooks/aws_glue_job_hook.py:
```diff
@@ -136,56 +136,59 @@ def job_completion(self, job_name=None, run_id=None):
             else:
-                self.log.info("Polling for AWS Glue Job {} current run state"
-                              .format(job_name))
+                self.log.info("Polling for AWS Glue Job {} current run state with status {}"
```
```suggestion
                self.log.info("Polling for AWS Glue Job %s current run state with status %s", job_name,job_run_state)

```
I think, you should avoid string formatting. 
Reference: https://github.com/apache/airflow/pull/4804

abdulbasitds replied:
@mik-laj can you please guideme,how can i submit my changes to airflow original repo. Auther of my fork isnt merging my pull request

mik-laj replied:
You must create a ticket in JIRA and change commit message. It should have following format: 
```
[AIRFLOW-1234] Commit message
````
In the next step, you can create a PR for Apache using the link below:
https://github.com/OElesin/incubator-airflow/compare/master...abdulbasitds:master

abdulbasitds committed:
Update airflow/contrib/hooks/aws_glue_job_hook.py

Co-Authored-By: Kamil Breguła <mik-laj@users.noreply.github.com>

abdulbasitds committed:
formatting error

abdulbasitds committed:
deleted the redundant hook file

abdulbasitds committed:
resolving test errors - pylint whitespaces

abdulbasitds committed:
resolving test errors - pylint

abdulbasitds committed:
Moving glue to provider.amazon.aws

abdulbasitds committed:
Added sensor/operator/hook for glue

abdulbasitds committed:
Moving tests to provider

abdulbasitds committed:
Rename test_aws_glue_job_sensor to test_aws_glue_job_sensor.py

abdulbasitds committed:
moving tests to provider

abdulbasitds committed:
moving tests to provider

abdulbasitds committed:
Delete test_aws_glue_job_operator.py

abdulbasitds committed:
Delete test_aws_glue_job_sensor.py

abdulbasitds committed:
Delete test_aws_glue_job_hook.py

abdulbasitds committed:
Delete code.rst

abdulbasitds committed:
Deleting contrib files

abdulbasitds committed:
Deleting changes from Integration.rst

abdulbasitds committed:
Delete test_aws_glue_job_sensor.py

abdulbasitds committed:
removing a file from PR

abdulbasitds committed:
Update airflow/providers/amazon/aws/operators/glue.py

empty line between documentation header and param section

Co-Authored-By: Felix Uellendall <feluelle@users.noreply.github.com>

abdulbasitds committed:
Update airflow/providers/amazon/aws/hooks/glue.py

Co-Authored-By: Felix Uellendall <feluelle@users.noreply.github.com>

abdulbasitds committed:
Update airflow/providers/amazon/aws/operators/glue.py

Co-Authored-By: Felix Uellendall <feluelle@users.noreply.github.com>

abdulbasitds committed:
resolving change requests for pull request

abdulbasitds committed:
corrected the reference to glue hook

abdulbasitds committed:
Implementing change requests

abdulbasitds committed:
Implementing change requests 2

abdulbasitds committed:
moved file upload to operator

kaxil committed:
Remove deprecated `secure_model` config (#8309)

* Remove deprecated `secure_model` config

`secure_mode` config was only available for Flask-admin UI

* fixup! Remove deprecated `secure_model` config

rcjsuen committed:
Add simple tests for SqliteOperator (#8307)

Improve code coverage by adding a few simple tests for the
SqliteOperator.

Signed-off-by: Remy Suen <remy.suen@gmail.com>

mik-laj committed:
Fix TOC on "How-to Guides/GCP" (#8295)

michalslowikowski00 committed:
fix typo in DAG Serialization documentation (#8317)

potiuk committed:
Production image is now built automatically in Dockerhub (#8314)

feluelle committed:
Expose Airflow Webserver Port in Production Docker Image (#8228)

mik-laj committed:
Reorder middleware - ProxyFix and BaseUrl (#8157)

turbaszek committed:
Use Github Actions to run CI (#8376)

* Use Github Actions to run CI

* Fix backports build

dimberman committed:
[AIRFLOW-6885] Change delete-on-success to delete-on-failure (#8312)

* Change delete-on-success to delete-on-failure

It makes more sense to by-default not delete failed pods
Users should explicitly state they want these pods deleted.

This feature has not yet been released so ethis will not be a breaking
change

* deps

* deps

Co-authored-by: Daniel Imberman <daniel@astronomer.io>

kaxil committed:
Fix Extra Links in Gannt View (#8308)

Extra link didn't appear after changes in  https://github.com/apache/airflow/pull/8220 for Gantt View

turbaszek committed:
Add Github Actions badge to README (#8386)

roelhogervorst committed:
Add Dataproc SparkR Example (#8240)

* GCP SparkR Example

Allows you to schedule R, and sparkR jobs on a dataproc cluster.
The functionality to run that kind of job is already in dataproc,
but it was not so clear how to do that from Airflow.

* Update airflow/providers/google/cloud/example_dags/example_dataproc.py

Co-Authored-By: Tomek Urbaszek <turbaszek@gmail.com>

* Make sure R file finds correct library

Co-authored-by: Tomek Urbaszek <turbaszek@gmail.com>

potiuk committed:
Move backport packages to GA (#8391)

dimberman committed:
Add migration waiting script and log cleaner (#8219)

* Add migration waiting script and log cleaner

This PR creates a "migration spinner" that allows the webserver to wait for all database migrations to complete before starting up. Is a necessary component before we can merge the helm chart.

* Update airflow/cli/cli_parser.py

Co-Authored-By: Tomek Urbaszek <turbaszek@gmail.com>

Co-authored-by: Tomek Urbaszek <tomasz.urbaszek@polidea.com>
Co-authored-by: Tomek Urbaszek <turbaszek@gmail.com>

mik-laj committed:
Display docs errors summary (#8392)

mik-laj committed:
Add airflow info command (#8290)

gnomeria committed:
Add qoala as officially using airflow (#8401)

We're using it as part of our ETL and other data related scheduling

nadflinn committed:
Handle no Dagrun in DagrunIdDep (#8389)

alexandrecaze committed:
[AIRFLOW-8187] Extend elastic DAG with a binary tree, grid, star (#8277)

ashb committed:
Fix timing-based flakey test in TestLocalTaskJob (#8405)

This test suffered from timing-based failures, if the "main" process
took even fractionally too long then the task process would have already
cleaned up it's subprocess, so the expected callback in the main/test
process would never be run.

This changes is so that the callback _will always be called_ in the test
process if it is called at all.

mik-laj committed:
Add colors to airflow config command (#8404)

potiuk committed:
Test script tries to reset db regardless of db check (#8403)

ashb committed:
Allow "falsey" default arguments in CLI Parser (#8398)

In #8219 we noticed that we couldn't set a `default=0` because of the
`and v` check. The "add_argument" function in python avoid this by using
**kwargs, but we want type checking so can't directly use the same
there.

This uses the same pattern that configparser does to allow falsey (0,
False) and even `None` as valid values, distinct from not being passed.

dimberman committed:
Fixes check_migrations commands (#8407)

Fixes bug where old "wait_for_migrations" function name
is left in the lazy loader of the cli_parser

Co-authored-by: Daniel Imberman <daniel@astronomer.io>

potiuk committed:
Fix building image manifest (was removed by accident with prod img) (#8408)

BasPH committed:
Fix Migration for MSSQL (#8385)

boittega committed:
Update the tree view of dag on Concepts Last Run Only (#8268)

Resolves #8246

mbarrien committed:
Fix error message in production entrypoint.sh (#8396)

Fix non-existent BACKEND environment variable, replace with DB_URL

potiuk committed:
Less frequent dockerhub builds (#8400)

mik-laj committed:
Clean up temporary files in Dataflow operators (#8313)

mik-laj committed:
Use less fancy tables in CLI by default (#8409)

kaxil committed:
Add a dedicated "free disk space" step to fix CI (#8426)

kaxil committed:
Make doc clearer about Airflow Variables using Environment Variables (#8427)

haoliang7 committed:
Fix subcommand error when running production image without argument (#8415)

Co-authored-by: Liang Hao <liahao@tesla.com>

Commit:
stop rendering some class docs in wrong place (#8095)

* stop rendering some class docs in wrong place

Docs generated for providers.yandex.hooks incorrectly include docs
for airflow.exceptions.AirflowException and
airflow.hooks.base_hook.BaseHook in the yandex module, as if
those classes had been defined in that module.

This is almost certainly a bug in autoapi or one of the libraries
it uses, but I haven't tracked down the root cause. In the meantime,
importing the modules and then referring to the classes using
modulename.Classname avoids the issue.

kaxil committed:
Remove duplicate dependency ('curl') from Dockerfile (#8412)

michalslowikowski00 committed:
fixed typo in confirm script (#8419)

Co-authored-by: michalslowikowski00 <michal.slowikowski@polidea.com>

kaxil committed:
Add back-compat modules from 1.10.10 for SecretsBackends (#8413)

randr97 committed:
[AIRFLOW-5156] Added auth type to HttpHook (#8429)

randr97 committed:
[AIRFLOW-5156] Fixed doc strigns for HttpHook (#8434)

danfrankj committed:
Fix performance degradation when updating dagrun state (#8435)

Co-authored-by: Dan Frank <dan.frank@coinbase.com>

edejong committed:
Added location parameter to BigQueryCheckOperator (#8273)

kaxil committed:
Fix formatting of Pool docs in concepts.rst (#8443)

potiuk committed:
Not pulling base python images does not work for bugfixes (#8437)

kaxil committed:
Improve language in Pod Mutation Hook docs (#8445)

kaxil committed:
Make KubernetesPodOperator clear in docs (#8444)

Avoid confusion with https://kubernetes.io/docs/concepts/extend-kubernetes/operator/

kaxil committed:
Fix formatting in Lineage docs (#8446)

turbaszek committed:
Simplify mocking in BigQueryHook tests (#8450)

ivorynoise committed:
[AIRFLOW-1536] Set a default value or read from cfg/args for setting DaemonContext in worker (#7724)

belidzs committed:
Added more precise Python requirements to README.md (#8455)

turbaszek committed:
Remove unittest.main() from tests (#8454)

mustafagok committed:
Remove unrelated EC2 references in ECSOperator (#8451)

kaxil committed:
Allow only adding labels on PR Creation (not Updation) (#8461)

mik-laj committed:
Allow multiple extra_packages in Dataflow (#8394)

Co-authored-by: Tomek Urbaszek <tomasz.urbaszek@polidea.com>

potiuk committed:
Fix case when the pidfile is empty during the pidfile check (#8462)

mik-laj committed:
Return non-zero error code when variable is missing(#8438)

mik-laj committed:
Divide commands into "Actions"/"Groups" sections (#8456)

mik-laj committed:
Allow retrieving Connections from Secrets Backend using CLI (#8440)

Co-Authored-By: Kaxil Naik <kaxilnaik@gmail.com>

dhuang committed:
Fix Snowflake hook conn id (#8423)

Do not set manually to ensure the base `DbApiHook` correctly interprets
arguments based on `conn_name_attr` and `default_conn_name`.

fhoda committed:
Add Airflow Users for Optum (#8478)

potiuk committed:
Pin Hadolint to version released 2020.04.20 (#8485)

haoliang7 committed:
Fix awkward log info in dbapi_hook (#8482)

Co-authored-by: Liang Hao <liahao@tesla.com>

potiuk committed:
Pre-commit checks in Github Actions have colour and are simpler (#8486)

We have now back color in pre-commit checks and we disabled verbose
flag. The main reason for verbose flag was to show execution time
of each commit, but this is not needed in Github Actions as we
have the "Show Timestamp" option in job log which gives
us all that we need. Also the id of failing pre-commit check
is displayed whenever it fails so we do not need --verbose flag
any more.

potiuk committed:
Update to latest pygrep pre-commit hook (#8489)

potiuk committed:
Get rid of Travis CI from the docs (#8488)

dinigo committed:
Fix typo in UPDATING.md (#8493)

kaxil committed:
Use existing DagBag for 'dag_details' & `trigger` Endpoints (#8501)

mik-laj committed:
Add scripts/list-integrations.py (#8469)

eladkal committed:
[AIRFLOW-7059] pass hive_conf to get_pandas_df in HiveServer2Hook (#8380)

* [AIRFLOW-7059] pass hive_conf to get_pandas_df in HiveServer2Hook

* [AIRFLOW-7059] pass hive_conf to get_pandas_df in HiveServer2Hook

* Use Github Actions to run CI (#8376)

* Use Github Actions to run CI

* Fix backports build

Co-authored-by: Tomek Urbaszek <turbaszek@gmail.com>

turbaszek committed:
Use python client in BQ hook create_empty_table/dataset and table_exists (#8377)

* Use python client in BQ hook create_empty_table method

* Refactor table_exists and create_empty_dataset

* Add note in UPDATING

potiuk committed:
WTForms 2.3.0 break our Flask apps (#8512)

* WTForms 2.3.0 break our Flask apps

kaxil committed:
Remove unused session variable from www/view.py (#8504)

potiuk committed:
List of integrations is now maintained in one place. (#8496)

potiuk committed:
Add installation description for repeatable PyPi installation (#8513)

potiuk committed:
Fix too long comment in setup.py (#8515)

Fixes Bad commit via Github "suggestion" #8512

turbaszek committed:
Add "please use search" in bugfix template (#8492)

* Please use search before creating bug issue

* fixup! Please use search before creating bug issue

potiuk committed:
Add support for caching of image in GitHub's registry (#8497)

kaxil committed:
Remove WTforms from setup.py (#8522)

potiuk committed:
Optimize GitLab CI configuration (#8499)

* tests are not executed for doc-only changes
* images will be (once merged) downloaded from GitHub Registry so likely much faster
* we have a "scheduled" nightly build that will build everything from scratch and check if no requirements have been broken
* improved split of static checks between two static check jobs - to utilise parallelism better.
* reorganised some fast jobs (requirements, prod image) that do not depend on tests so that they can run earlier
* shorter names for jobs so that they are nicer to view in the actions view
* matrix definitions of the jobs so that we can manage them better

potiuk committed:
Move some tests to quarantine (#8511)

michalslowikowski00 committed:
Added GoogleDisplayVideo360DownloadLineItemsOperator (#8174)

Co-authored-by: michalslowikowski00 <michal.slowikowski@polidea.com>

mik-laj committed:
Pass location using parmamter in Dataflow integration (#8382)

tatitati committed:
Add Simply Business to the documentation-list of "companies using Airflow" (#8516)

* moved to a better place

* renamed name

zhongjiajie committed:
Development PR template target issue if exists (#8524)

ryanahamilton committed:
[AIRFLOW-4357] Fix SVG tooltip positioning with custom scripting (#8269)

johnnybravo-xyz committed:
Add Zerodha to Airflow users list (#8532)

Commit:
[AIRFLOW-8474]: Adding possibility to get job_id from Databricks run (#8475)

* [AIRFLOW-8474]: Adding possibility to get job_id from Databricks run

Co-Authored-By: Jiajie Zhong <zhongjiajie955@hotmail.com>

jcarless committed:
[AIRFLOW-7111] Add generate_presigned_url method to S3Hook (#8441)

Commit:
fix help message display for dags test subcommand (#8552)

feluelle committed:
Fix --forward-credentials flag in Breeze (#8554)

ephraimbuddy committed:
Add hook and operator for Google Cloud Life Sciences (#8481)

potiuk committed:
The CRON job now is working and triggers builds on DockerHub (#8549)

The CRON job from previous runs did not have everything working
after the emergency migration to Github Actions.

This change brings back following improvements:

* rebuilding images from the scratch in CRON job
* automatically upgrading all requirements to test if they are new
* pushing production images to github packages as cache
* pushing nightly tag to github

Commit:
Add test to guard against command arg help message regression (#8561)

follow up for PR #8552.

wkhudgins92 committed:
[AIRFLOW-6281] Create guide for GCS to GCS transfer operators (#8442)

mustafagok committed:
Add example DAG for ECSOperator (#8452)

potiuk committed:
This test (test_mark_success_on_success_callback) fails often (#8563)

I am moving the test to quarantine for now.

potiuk committed:
Retry initialization of environment in case it fails (#8555)

Sometimes the docker-compose environment fails to initialise. It is
one of two problems:

* DNS fwd/rev mismatch on checking the docker compose integration
* Missing name for the docker compose intagration

This is likely due to some race conditions when setting up docker
compose network and DNS. It happens very rarely and it seems that
it's not easily reproducible. It is very likely that detecting
this case, cleaning the docker environment and retrying will
work fine. This is what this PR tries to attempt

potiuk committed:
Building backport packages is now done inside Breeze container (#8558)

This way you do not have to worry about setting up your environment.

Fixes #8537

potiuk committed:
Fixed SOURCE_BRANCH set by tag names (#8562)

SOURCE_BRANCH was set to TAG_NAME in Dockerhub. Now that we changed
DockerHub to build from tags, it caused AIRFLOW_BRANCH arg
to be set to master-nightly instead of master and we had a cache
miss.

SOURCE_BRANCH variable is not realy needed any more. Removed it

khyurri committed:
Improve idempodency in CloudDataTransferServiceCreateJobOperator (#8430)

* Fix issue #8285

- add JOB_NAME to available request body keys
- check error on creating new transfer
- add tests
- add gen_job_name function, that generates suffix if name was deleted (this behavior is not completed)
- add method enable_transfer_job
- add gen_job_name function, that generates suffix if job was deleted
- change CloudDataTransferServiceCreateJobOperator documentation
- pylint minor fix
- code review fix on gen_job_name function
- Fix build docs error
- Rename some inner variables for pylint
- Remove unused import
- Stylefix after precommit
- refactor test
- refactor gen_job_name() method

akki committed:
[AIRFLOW-5850] Capture task logs in DockerSwarmOperator (#6552)

* [AIRFLOW-5850] Capture task logs in DockerSwarmOperator

* [AIRFLOW-5850] Fix the mock in the docker swarm tests

* [AIRFLOW-5850] Squash me: Remove nested blocks in docker swarm operator

mppetkov committed:
AIRFLOW-6062 Watch worker pods from all namespaces (#8546)

aamangeldi committed:
Monkey patch greenlet celery pools (#8559)

Celery pools of type eventlet and gevent use greenlets,
which requires monkey patching the app:
https://eventlet.net/doc/patching.html#monkey-patch

Otherwise task instances hang on the workers and are never
executed.

potiuk committed:
File should be specified in docker compose before command (#8569)

potiuk committed:
Add test_localtaskjob_maintain_heart_rate to quarantine (#8566)

OmairK committed:
[AIRFLOW-4438] Add Gzip compression to S3_hook (#8571)

Fixes a bug introduced in #7680 with passing filename as string

mik-laj committed:
User-friendly error messages when the configuration is incorrect (#8463)

* Clearer error messages when the configuration is incorrect

potiuk committed:
Bash command for production image (#8579)

mik-laj committed:
Add Local Filesystem Secret Backend (#8436)

Commit:
Fix json string escape in tree view (#8551)

close #8523.

potiuk committed:
Update documentation with clarification about v1-10-stable (#8588)

The v1-10-test branch does not have it's own cron job. The
DockerHub build is triggered after manual pushing of v1-10-stable.

kaxil committed:
Pin Pylint to 2.4.4 (#8592)

Pylint 2.5.0 was released which has added bunch of new rules & checks.

Pinning to 2.4.4 till we check the new rules & disable the ones we don't want to use

MatthewRBruce committed:
[AIRFLOW-6796] Clean up DAG serializations based on last_updated (#7424)

DAG serializations were previous deleted based on whether the
DagFileProcessorManager had processed a particular python file.  This
changes that to be based on the last time a DAG was processed by the
scheduler.

Also moves cleaning up of stale dags to the DagFileProcessorManager to
support long running schedulers

kaxil committed:
Remove WTforms from setup.py (#8590)

mik-laj committed:
Fix typo in error message - systems -> system (#8584)

Commit:
[AIRFLOW-8472]: `PATCH` for Databricks hook `_do_api_call` (#8473)

Support of `PATCH` for Databricks hook `_do_api_call` function

CodingJonas committed:
Stop DockerSwarmOperator from pulling Docker images (#8533)

Co-authored-by: Jonas Natzer <jonas.natzer@celus.io>

kaxil committed:
Add back-compat secrets to TestMovingCoreToContrib (#8593)

Commit:
fix: aws hook should work without conn id (#8534)

This patch makes behavior of hook consistent with documentation.

AWS hooks should support falling back to using default credential chain
lookup behavior when connection id is not specified.
add test for conn_id equals None
more elegant way to set role session name

mik-laj committed:
Fix example for Local Filesystem Secrets Backend (#8597)

potiuk committed:
Fixed optimistions of non-py-code builds (#8601)

Count of changed files returned exit code 1 not 0 when there
was no files matching the pattern, so the whole optimisation
did not work as intended.

potiuk committed:
Chown should work now when building the documentation (#8600)

turbaszek committed:
Refactor BigQueryHook dataset operations (#8477)

* Refactor database methods of BigQueryHook

* fixup! Refactor database methods of BigQueryHook

* Apply suggestions from code review

Co-Authored-By: Michał Słowikowski <michalslowikowski00@gmail.com>

* fixup! fixup! Refactor database methods of BigQueryHook

* fixup! fixup! fixup! Refactor database methods of BigQueryHook

* Pass credentials to BQ client

Co-authored-by: Michał Słowikowski <michalslowikowski00@gmail.com>

turbaszek committed:
Split and improve BigQuery example DAG (#8529)

* Simplify existing example DAG and add consistent UI colors

* Move BigQueryConsoleLink to top

* Split DAG into two separate

* Add transfer example

* Fix check interval operator

Could not cast literal {{ macros.ds_add(ds, -1) }} to type DATE at [1:51]'}], 'state': 'DONE'}}

* Add BigQuery queries example

* fixup! Add BigQuery queries example

turbaszek committed:
Allow to define custom XCom class (#8560)

* Allow to define custom XCom class

closes: #8059

kaxil committed:
Pin Flask-Appbuilder to 2.3.2 (#8602)

feluelle committed:
Fix sphinx add_stylesheet deprecation warning (#8627)

See https://www.sphinx-doc.org/en/master/extdev/appapi.html#sphinx.application.Sphinx.add_css_file for more information.

feluelle committed:
Add http system test (#8591)

Commit:
[AIRFLOW-4734] Upsert functionality for PostgresHook.insert_rows() (#8625)

PostgresHook's parent class, DbApiHook, implements upsert in its insert_rows() method
with the replace=True flag. However, the underlying generated SQL is specific to MySQL's
"REPLACE INTO" syntax and is not applicable to PostgreSQL.

This pulls out the sql generation code for insert/upsert out in to a method that is then
overridden in the PostgreSQL subclass to generate the "INSERT ... ON CONFLICT DO
UPDATE" syntax ("new" since Postgres 9.5)

potiuk committed:
Fix the process of requirements generations (#8648)

Right now requirements will be only checked during the
CI build if the setup.py has changed and if yes, clear instructions
will be given. The diff will still be printed otherwise but
it will not cause the job to fail

XD-DENG committed:
Reduce response payload size of /dag_stats and /task_stats (#8633)

Their response format is like {"example_dag_id": [{"state": "success", "dag_id": "example_dag_id"}, ...], ...}

The dag_id is already used as the "key", but still repeatedly appear in each element,
which makes the response payload size unnecessarily bigger

oripwk committed:
Improve template capabilities of EMR job and step operators (#8572)

Allow EmrCreateJobFlowOperator and EmrAddStepsOperator
to receive their 'job_flow_overrides', and 'steps'
arguments respectively as Jinja template filenames.
This is similar to BashOperator's capability of
receiving a filename as its 'bash_command' argument.

vardancse committed:
Enhanced documentation around Cluster Policy (#8661)

retornam committed:
[AIRFLOW-4363] Fix JSON encoding error (#8287)

dimberman committed:
Add check for pre-2.0 style hostname_callable config value (#8637)

Changes deprecated config check rules. Now uses regex to look for an old pattern in the val. Updates 'hostname_callable'.

This lets us pull the change back in to 1.10.x, so that by the time 2.0 is around people will have had time and notice to update, without reading (the now quite long) UPDATING.md.

Depends on #8463

kaxil committed:
Fix displaying Executor Class Name in "Base Job" table (#8679)

XD-DENG committed:
Persist start/end date and duration for DummyOperator Task Instance (#8663)

Otherwise the behaviour in UI is incorrect
Addressing issue https://github.com/apache/airflow/issues/8662

XD-DENG committed:
Ensure "started"/"ended" in tooltips are not shown if job not started (#8667)

msumit committed:
Add support for fetching logs from running pods (#8626)

When using KubernetesExecutor without any centralized PV for log storage, one has to wait until the logs get uploaded to cloud storage before viewing them on UI. With this change, the webserver will try to fetch logs from running worker pods and display them.

fusuiyi123 committed:
Remove _get_pretty_exception_message in PrestoHook

Fix #8530

Lyalpha committed:
Improve tutorial - Include all imports statements (#8670)

mik-laj committed:
Group Google services in one section (#8623)

Commit:
Refactor test_variable_command.py (#8535)

randr97 committed:
Add system test and docs for Facebook Ads operators (#8503)

XD-DENG committed:
Fix connection add/edit for spark (#8685)

connection add/edit UI pages were not working correctly for Spark connections. 

The root-cause is that "spark" is not listed in models.Connection._types.
So when www/forms.py tries to produce the UI,
"spark" is not available and it always tried to "fall back" to the option list
whose first entry is "Docker"

In addition, we should hide irrelevant entries for
spark connections ("schema", "login", and "password")

XD-DENG committed:
Sort connection type list in add/edit page alphabetically (#8692)

Currently the connection type list in the UI is sorted in the original order of
`Connection._types`, which may be a bit inconvenient for users.

It would be better if it can be sorted alphabetically.

Commit:
Support k8s auth method in Vault Secrets provider (#8640)

joppevos committed:
Add system test for gcs_to_bigquery (#8556)

mik-laj committed:
[AIRFLOW-7008] Add perf kit with common used decorators/contexts (#7650)

mik-laj committed:
Invalid output in test_variable assertion (#8698)

mik-laj committed:
Change provider:GCP to provider:Google for Labeler Bot (#8697)

turbaszek committed:
Remove config side effects from tests (#8607)

* Remove config side effects

* Fix LatestOnlyOperator return type to be json serializable

* Fix tests/test_configuration.py

* Fix tests/executors/test_dask_executor.py

* Fix tests/jobs/test_scheduler_job.py

* Fix tests/models/test_cleartasks.py

* Fix tests/models/test_taskinstance.py

* Fix tests/models/test_xcom.py

* Fix tests/security/test_kerberos.py

* Fix tests/test_configuration.py

* Fix tests/test_logging_config.py

* Fix tests/utils/test_dag_processing.py

* Apply isort

* Fix tests/utils/test_email.py

* Fix tests/utils/test_task_handler_with_custom_formatter.py

* Fix tests/www/api/experimental/test_kerberos_endpoints.py

* Fix tests/www/test_views.py

* Code refactor

* Fix tests/www/api/experimental/test_kerberos_endpoints.py

* Fix requirements

* fixup! Fix tests/www/test_views.py

mik-laj committed:
Check consistency between the reference list and howto directory (#8690)

Acehaidrey committed:
Prevent clickable sorting on non sortable columns in TI view (#8681)

Co-authored-by: Ace Haidrey <ahaidrey@pinterest.com>

jhtimmins committed:
Import Connection directly from multiprocessing.connection. (#8711)

Co-authored-by: James Timmins <james@astronomer.io>

michalslowikowski00 committed:
Fix typo in Google Display & Video 360 guide

Co-authored-by: michalslowikowski00 <michal.slowikowski@polidea.com>

mik-laj committed:
Carefully parse warning messages when building documentation (#8693)

mik-laj committed:
Support num_retries field in env var for GCP connection (#8700)

XD-DENG committed:
Add __repr__ for DagTag so tags display properly in /dagmodel/show (#8719)

Commit:
Add guide for Apache Spark operators (#8305)

jhtimmins committed:
Fix pickling failure when spawning processes (#8671)

* Pull processor_factory out of _execute and move to the class scope.

* Change the value of pickle_dags from True to False which is the default value of pickle_dags to be passed.

* Fix how to reference FakeDagFileProcessorRunner class due to the place where it's defined is changed.

* Fix configuration inheritance issue using  multiprocessing with spawn mode.
* Add a new CI entry for spawn-multiprocessing method.

* Add testcases for multiprocessing with spawn mode.

Co-authored-by: Kousuke Saruta <sarutak@oss.nttdata.com>
Co-authored-by: James Timmins <james@astronomer.io>

mik-laj committed:
Support all RuntimeEnvironment parameters in DataflowTemplatedJobStartOperator (#8531)

mik-laj committed:
Add  jinja template test for AirflowVersion (#8505)

mik-laj committed:
Avoid loading executors in jobs (#7888)

mik-laj committed:
Optimize count query on /home (#8729)

ashb committed:
Correctly deserialize dagrun_timeout field on DAGs (#8735)

We weren't deserializing this correctly (it was left as a float) but
nothing _was_ using it, and we hadn't explicitly tested it.

We already have example dags with this field, so we just need to check
for this field.

kaxil committed:
Stop Stalebot on Github issues (#8738)

kaxil committed:
Make loading plugins from entrypoint fault-tolerant (#8732)

jhtimmins committed:
Ensure test_logging_config.test_reload_module works in spawn mode. (#8741)

Co-authored-by: James Timmins <james@astronomer.io>

potiuk committed:
Latest debian-buster release broke image build (#8758)

feluelle committed:
Add google_api_to_s3_transfer example dags and system tests (#8581)

- add amazon system helper for easier testing amazon aws systems / services
- fix TESTING docs

feluelle committed:
Add google_api_to_s3_transfer docs howto link (#8761)

ephraimbuddy committed:
Add documentation for SpannerDeployInstanceOperator (#8750)

lokeshlal committed:
[AIRFLOW-4568]The ExternalTaskSensor should be configurable to raise an Airflow Exception in case the poked external task reaches a disallowed state, such as f.i. failed (#8509)

Added failed_states for ExternalTaskSensor to avoid waiting for the failure scenarios till timeout value

mik-laj committed:
Add SQL query tracking for pytest (#8754)

abdulbasitds committed:
Reslving all unit and stating checks and also comments

michalslowikowski00 committed:
Added SDFtoGCSOperator (#8740)

Co-authored-by: michalslowikowski00 <michal.slowikowski@polidea.com>

vshshjn7 committed:
Patch Pool.DEFAULT_POOL_NAME in BaseOperator (#8587)


Co-authored-by: Vishesh Jain <visheshj@twitter.com>

ashb committed:
Support same short flags for `create user` as 1.10 did for `user_create` (#8763)

This will make it easier to transition between versions.

* fixup! Support same short flags for `create user` as 1.10 did for `user_create`

kaxil committed:
Update airflow/providers/amazon/aws/operators/glue.py

kaxil committed:
Fix failing doc test

Commit:
Add WorldRemit as Airflow user (#8786)

Commit:
fix typing errors reported by dmypy (#8773)

ashb committed:
Update example SingularityOperator DAG (#8790)

The main thing I was fixning here was `start_date=utcnow()` which is
always going to be wrong (discovered via a test in #8772).

While I was updating the DAG I updated it to use context manager, and
shift operators.

potiuk committed:
Backport packages are renamed to include backport in their name (#8767)

potiuk committed:
Fixed test-target command (#8795)

Wrong parameter sequence were passed to the run script.

nadflinn committed:
Make celery worker_prefetch_multiplier configurable (#8695)

abdulbasitds committed:
remaining changes

kaxil committed:
Update docs/operators-and-hooks-ref.rst

kaxil committed:
Update tests/providers/amazon/aws/hooks/test_aws_glue_job_hook.py

abdulbasitds committed:
resolving last errors

abdulbasitds committed:
Merge branch 'master' of https://github.com/abdulbasitds/incubator-airflow

abdulbasitds committed:
reolving flake space issue

abdulbasitds committed:
removing connection and renaming test files

abdulbasitds committed:
test files missing error resolution

abdulbasitds committed:
catering timout and failure exception in hook

abdulbasitds committed:
Trigger Github actions

abdulbasitds committed:
flake error and job_interval constant removal
